File: android/src/main/kotlin/com/yd/momo_payment_plugin/MomoPaymentResult.kt
package com.yd.momo_payment_plugin

internal object MomoPaymentResults {

    fun error(status: Int): Map<String, Any> {
        return HashMap<String, Any>().apply {
            put("isSuccess", false)
            put("status", status)
        }
    }

    fun success(token: String, phonenumber: String): Map<String, Any> {
        return HashMap<String, Any>().apply {
            put("isSuccess", true)
            put("status", 0)
            put("token", token)
            put("phonenumber", phonenumber)
        }
    }
}

File: lib/momo_payment_result.dart
class MomoPaymentResult {
  bool isSuccess;
  int status;
  String token;
  String phonenumber;

  MomoPaymentResult({
    this.isSuccess,
    this.status,
    this.token,
    this.phonenumber,
  });

  static MomoPaymentResult fromMap(Map<String, dynamic> map) {
    return MomoPaymentResult(
      isSuccess: map['isSuccess'],
      status: map['status'] != null ? map['status'] : null,
      token: map['token'] != null ? map['token'] : null,
      phonenumber: map['phonenumber'] != null ? map['phonenumber'] : null,
    );
  }
}
